docs(apisix): wrote `mTLS between APISIX and Upstream` (#156)
api7/docs: apisix_versioned_docs/version-3.3.0/how-to-guide/traffic-management/tls-and-mtls/enable-upstream-mtls.md

---
title: Enable Upstream mTLS
slug: /how-to-guide/traffic-management/tls-and-mtls/enable-upstream-mtls
---

_Mutual TLS_ (also known as mTLS) is a method for mutual authentication. It is an extension of the standard TLS protocol to make sure that both the client and the server must authenticate each other.

mTLS between an API gateway and its upstream services is typically implemented in high-security environments by organizations, such as financial institutions, who need to stay compliant with relevant security standards and regulations.

This guide will configure an NGINX server with mTLS as an upstream service, then show you how to configure mTLS between APISIX and an upstream service.

<br/>
  <div style={{textAlign: 'center'}}>
    <img src="https://static.apiseven.com/uploads/2023/05/09/sNYhwgBS_mtls-apisix-upstream.jpg" alt="TLS between APISIX and Upstream" width="70%"/>
  </div>
<br/>

## Prerequisite(s)

* Install [Docker](https://docs.docker.com/get-docker/).
* Install [cURL](https://curl.se/) to send requests to the service for test.
* Follow the [Getting Started tutorial](../../../getting-started/) to start a new APISIX instance in Docker.

## Create Certificates

> Create root _certificate authority_ (also known as CA) certificate private key `ca.key` and public key `ca.crt` to sign both the client and server certificates:

Reviewer: ```suggestion
Create a [root CA](https://en.wikipedia.org/wiki/Root_certificate) certificate `ca.crt` with the private key `ca.key` and a [certificate signing request](https://en.wikipedia.org/wiki/Certificate_signing_request) `ca.csr`:
```

Author: Do we need to link to the 3rd webpage such as wikipedia?

```shell
openssl genrsa -out ca.key 2048
openssl req -new -sha256 -key ca.key -out ca.csr -subj "/CN=ROOTCA"
openssl x509 -req -days 36500 -sha256 -extensions v3_ca -signkey ca.key -in ca.csr -out ca.crt
```

> Create server certificate private key `server.key` and public key `server.crt` with the _common name_ (also known as CN) `test.com`:

Reviewer: ```suggestion
Create a private key `server.key` and a server certificate `server.crt` with common name (CN) `test.com`:
```

```shell
openssl genrsa -out server.key 2048
openssl req -new -sha256 -key server.key -out server.csr -subj "/CN=test.com"
openssl x509 -req -days 36500 -sha256 -extensions v3_req -CA ca.crt -CAkey ca.key -CAserial ca.srl -CAcreateserial -in server.csr -out server.crt
```

> Create client certificate private key `client.key` and public key `client.crt`:

Reviewer: ```suggestion
Create a client private key `client.key` and a client certificate `client.crt` with common name (CN) `CLIENT`:
```
